Sources/Helpers/Vendor/_AsyncFileSystem/SDKFileSystem.swift

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023-2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@preconcurrency import _NIOFileSystem
+
+public actor SDKFileSystem: AsyncFileSystem {
+    public init(readChunkSize: Int = defaultChunkSize) {
+        self.readChunkSize = readChunkSize
+    }
+    public static let defaultChunkSize = 512 * 1024
+    let readChunkSize: Int
+    public func exists(_ path: SystemPackage.FilePath) async -> Bool {
+        do {
+            guard let _ = try await FileSystem.shared.info(forFileAt: path) else {
+                return false
+            }
+            return true
+        } catch {
+            return false
+        }
+    }
+
+    public func withOpenReadableFile<T: Sendable>(
```

Comment:
I don't think we should reinvent the wheel here and reimplement `with*` methods from scratch. The main motivation for using `_NIOFileSystem` in the first place is that it already [implements these methods](https://swiftpackageindex.com/apple/swift-nio/2.73.0/documentation/_niofilesystem/filesystem/withfilehandle(forreadingandwritingat:options:execute:)) and provides necessary abstractions, like [`FileSystemProtocol`](https://swiftpackageindex.com/apple/swift-nio/2.73.0/documentation/_niofilesystem/filesystemprotocol).